processing_r/processing/algorithm.py

```diff
@@ -456,13 +465,22 @@ def build_script_header_commands(self, _, __, ___):
             path_to_use = str(RUtils.r_library_folder()).replace('\\', '/')
             commands.append(self.r_templates.change_libPath(path_to_use))
 
-        packages = RUtils.get_required_packages(self.script)
-        packages.extend(self.r_templates.get_necessary_packages())
+        packages = self.r_templates.get_necessary_packages()
 
         for p in packages:
             commands.append(self.r_templates.check_package_availability(p))
             commands.append(self.r_templates.load_package(p))
 
+        if self.r_templates.install_github:
+            for dependency in self.r_templates.github_dependencies:
+                commands.append(self.r_templates.install_package_github(dependency))
+
+        packages_script = RUtils.get_required_packages(self.script)
+
+        for p in packages_script:
+            commands.append(self.r_templates.check_package_availability(p))
+            commands.append(self.r_templates.load_package(p))
+
```

Comment:
I think this whole block of code could be moved to r_templates -- what do you think?

Reply:
That makes sense, all the info is contained there anyway! I will fix it later today :)

Btw: Just out of curiosity, is there a way to add the changes to this pull request or do I have to close it and make a new one?

Reply:
If you commit them to the same branch and then push, they'll appear here automatically!